nlpsandboxclient/api_client.py

```diff
@@ -86,120 +98,182 @@ def _build_uri(self, uri, endpoint=None):
         """Lists all datasets"""
-        return self.rest_get_paginated("/datasets")
+        datasets = self.rest_get_paginated("/datasets")
+        for dataset in datasets:
+            # The id of the dataset is found in datasets/{datasetId}
+            # Which is the basename
+            yield Dataset(id=os.path.basename(dataset['name']), **dataset)
 
-    def get_dataset(self, datasetid: str):
+    def get_dataset(self, datasetid: str) -> Dataset:
         """Get a dataset"""
-        return self.rest_get(f"/datasets/{datasetid}")
+        dataset = self.rest_get(f"/datasets/{datasetid}")
+        return Dataset(id=datasetid, **dataset)
 
-    def create_dataset(self, datasetid: str):
+    def create_dataset(self, datasetid: str) -> Dataset:
         """Create a dataset"""
-        return self.rest_post(f"/datasets?datasetId={datasetid}",
-                              body=json.dumps({}))
+        dataset = self.rest_post(f"/datasets?datasetId={datasetid}",
+                                 body={})
+        return Dataset(id=datasetid, **dataset)
 
-    def list_annotation_stores(self, datasetid: str):
+    def list_annotation_stores(self,
+                               datasetid: str) -> Iterator[AnnotationStore]:
         """List the annotation stores for a dataset"""
-        return self.rest_get_paginated(
+        annotation_stores = self.rest_get_paginated(
             f"/datasets/{datasetid}/annotationStores"
         )
+        for store in annotation_stores:
+            yield AnnotationStore(datasetid=datasetid,
+                                  id=os.path.basename(store['name']),
+                                  **store)
 
-    def get_annotation_store(self, datasetid: str, annotation_storeid: str):
+    def get_annotation_store(self, datasetid: str,
+                             annotation_storeid: str) -> AnnotationStore:
         """Get an annotation store"""
-        return self.rest_get(
+        store = self.rest_get(
             f"/datasets/{datasetid}/annotationStores/{annotation_storeid}"
         )
+        return AnnotationStore(datasetid=datasetid, id=annotation_storeid,
+                               **store)
 
     def create_annotation_store(self, datasetid: str,
-                                annotation_storeid: str):
+                                annotation_storeid: str) -> AnnotationStore:
         """Create an annotation store"""
-        return self.rest_post(
+        store = self.rest_post(
             f"/datasets/{datasetid}/annotationStores?"
             f"annotationStoreId={annotation_storeid}",
-            body=json.dumps({})
+            body={}
         )
+        return AnnotationStore(datasetid=datasetid, id=annotation_storeid,
+                               **store)
 
-    def list_annotations(self, datasetid: str, annotation_storeid: str):
+    def list_annotations(self, datasetid: str,
+                         annotation_storeid: str) -> Iterator[Annotation]:
```

Comment:
Replace `annotation_storeid` by `annotation_store_id` that the code generator / other services that we have

Reply:
Apply this convention everywhere in the client

Reply:
The reason the code generator separate id from the rest is because I separate it in the schemas, for example `patientId`. The code generator converts camel case to snake case, which results in `patient_id`. If you ever need to do this conversion in a programmatic way, you can use this code I found: https://github.com/nlpsandbox/data-node/blob/develop/server/openapi_server/util.py#L145-L192

Reply:
Will address this issue #49 in another PR